src/os_joinrate.c

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright (c) 2021 notsurewhattoputhere
```

Comment:
I don't know copyright law, but if I put a name for me here, can it just be "jess" with my email on it?